### PR TITLE
reorganize launch files to isolate gazebo dependency

### PR DESCRIPTION
**Pull request by alsora**
Signed-off-by: Alberto Soragna <alberto.soragna@gmail.com>

## Description

This PR re-organizes the launch files, fixing minor errors and moving to a more modular organization.
In particular:
 - Move the `OffsetParser` class to the "main" launch file. This tool is needed only if we want to spawn the dock at the robot location. This knowledge shouldn't belong to the dock-specific launch file.
 - Use more descriptive names for launch arguments
 - Split the `rviz.py` launch file: this file was responsible both for starting rviz as well as for running the robot state publishers. These state publishers are needed even if we don't want rviz. Now they have been moved to a `robot_description.launch.py` file.
 - Split the `dock.py` launch file: this file was both defining the dock links as well as spawning the gazebo model. Now the gazebo-related part has been moved out to the common gazebo file, next to where we spawn the robot model.

## How Has This Been Tested?

Run empty world and AWS house simulations.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: irobot_create_bringup/launch/rviz2.launch.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+from launch.substitutions import PathJoinSubstitution
+from launch_ros.actions import Node
+
+
+def generate_launch_description():
+    pkg_create3_description = get_package_share_directory('irobot_create_description')
+
+    # Rviz
+    rviz_config_dir = PathJoinSubstitution([pkg_create3_description, 'rviz', 'model.rviz'])
+    rviz_logo_dir = PathJoinSubstitution([pkg_create3_description, 'rviz', 'irobot_logo.jpg'])
```
Comment (eborghi10) [inline irobot_create_bringup/launch/rviz2.launch.py:17]: Should these files be moved to `irobot_create_bringup`?

Comment (alsora) [reply irobot_create_bringup/launch/rviz2.launch.py:17]: I thought about this and I looked at  some other ROS repositories as example.
I found that the rviz config is usually located in the description package.

Maybe I should move only the irobot logo image.

**Comment by rjcausarano** [comment]
If you launch directly dock_description.launch.py or robot_description.launch.py or rviz2.launch.py do you get any errors?? In the past we had users reporting errors after they ran these launch files directly (even though they aren't meant to be ran directly but rather included in the main launch file).

**Comment by alsora** [comment]
@rjcausarano yes they work if I start them directly (obviously they are not very interesting)

**Review by rjcausarano — APPROVED** [review]

File: irobot_create_bringup/launch/rviz2.launch.py (hunk 1 above)
Comment (alsora) [reply irobot_create_bringup/launch/rviz2.launch.py]: I ended up moving both files to the `irobot_create_bringup` package.
The `.rviz` file simply defines the default widgets when you open rviz, so it's not related to the actual robot description.

**Comment by alsora** [comment]
Can I get a second approval?

**Review by justinIRBT — APPROVED** [review]

**Review by rjcausarano — APPROVED** [review]
LGTM

**Pull request merged**